File: ios/vansample/vansample/ViewController.m
#import "ViewController.h"
#import <MolocoVANSDK/UserInfo.h>

@interface ViewController ()

@end

@implementation ViewController
@synthesize textView;

- (void)viewDidLoad {
	[super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
    
    // Set app name only if you want to set app name different from your bundle name.
	[[MolocoEntryPoint sharedInstance] setAppName:@"MolocoVanSample"];
}

- (void)didReceiveMemoryWarning {
	[super didReceiveMemoryWarning];
	// Dispose of any resources that can be recreated.
}

- (void)handleVANResponse:(NSString *)response {
    // Handle response from VAN server.
	NSLog(@"Response: %@",response);
	[textView setText:[NSString stringWithFormat:@"Response: %@", response]];

    // You can also access device and user information that VAN collected.
    UserInfo *userInfo = [UserInfo sharedInstance];
    NSLog(@"Response: %@\nIDFA: %@\nModel: %@\nCarrier: %@", response, [userInfo idfa], [userInfo model], [userInfo carrier]);
}

-(IBAction)clickSendBtn:(id)sender {
	NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
	[dict setValue:@"moloco" forKey:@"user"];
	[dict setValue:@"31" forKey:@"age"];
	[dict setValue:@"0.124313" forKey:@"latitude"];
	[dict setValue:@"0.717662" forKey:@"longitude"];
	[MolocoEntryPoint sendEvent:@"ClickEvent" dataMap:dict delegate:self];
}

@end

File: ios/VanSampleSwift/VanSampleSwift/ViewController.swift
//
//  ViewController.swift
//  VanSampleSwift
//
//  Created by Hyeonseo Ku on 7/23/16.
//  Copyright © 2016 Moloco, Inc. All rights reserved.
//

import UIKit

class ViewController: UIViewController, MolocoApiCallback {
    @IBOutlet weak var textView: UITextView!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        // Set app name only if you want to set app name different from your bundle name.
        MolocoEntryPoint.setAppName("MolocoVanSwiftSample")
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func sendEvent(_ sender: UIButton) {
        let dataMap: NSMutableDictionary = [
            "user": "moloco",
            "age": "31",
            "latitude": "0.124313",
            "longitude": "0.717662"
        ]
        MolocoEntryPoint.sendEvent("ClickEvent", dataMap: dataMap as [NSObject: AnyObject], delegate: self)
    }

    func handleVANResponse(_ response: String!) {
        // Handle response from VAN server.
        textView.text = "Response = \(response)"
        
        // You can also access device and user information that VAN collected.
        let userInfo = UserInfo.sharedInstance() as AnyObject?
        print("Response = \(response)\nIDFA = \(userInfo?.idfa)\nOS = \(userInfo?.os)\nCarrier = \(userInfo?.carrier)\n")
    }

}

